src/commandHandlers/iam.ts

```diff
@@ -7,36 +7,67 @@ import { getUserRoles } from './guild.service';
 import { log } from '../logger';
 
 /**
- * This command assigns the role given in the argument to the user that executed the command, if that role can be
- * self-assigned.
+ * This command assigns or removes  the role given in the argument to the user that executed the command,
+ * if that role can be self-assigned.
  */
 export const command = async (
   arg: [string, string],
   embed: MessageEmbed,
   message: Message
 ) => {
-  const args = arg[1].toLowerCase().split(',');
+  // Check for add or remove keywords
+  const argCheck = arg[1].toLowerCase().split(' ');
+  let toAdd;
+
+  if (argCheck[0] === 'add') {
+    toAdd = true;
+  } else if (argCheck[0] === 'remove') {
+    toAdd = false;
+  } else {
+    return buildErrorEmbed('Missing arguments');
+  }
+
+  // get list of roles to modify
+  argCheck.shift();
+  const args = argCheck.join(' ').split(',');
+
   // Check if the user provided the role argument
   if (!args) {
     return buildErrorEmbed('Missing arguments');
   }
 
-  const rolesToAssign = args.map((x) => x.trim());
+  const rolesToModify = args.map((x) => x.trim());
 
-  for (const roleToAssign of rolesToAssign) {
+  for (const roleToModify of rolesToModify) {
     // Check if the provided role is self-assignable
-    if (selfAssignableRoles.find((role) => role === roleToAssign)) {
+    if (selfAssignableRoles.find((role) => role === roleToModify)) {
       const role = message.guild!.roles.cache.find(
-        (r) => r.name === roleToAssign
+        (r) => r.name === roleToModify
       );
       if (!role) {
-        log.error(`ERROR: Couldn't get the role: ${roleToAssign}`);
+        log.error(`ERROR: Couldn't get the role: ${roleToModify}`);
         return buildErrorEmbed();
       }
-      // Assign the role to the user of the message
+
       const member = message.member;
-      await member!.roles.add(role);
-      await tryAddOpenSourceUserSubscription(roleToAssign, message);
+      if (toAdd) {
+        if (member!.roles.cache.find((r) => r.name === roleToModify)) {
+          // Checks if the user already has the role
```

Comment:
maybe the comment should be above the if statement, what do you think?
```suggestion
        // Checks if the user already has the role
        if (member!.roles.cache.find((r) => r.name === roleToModify)) {
```

Reply:
that would be better..